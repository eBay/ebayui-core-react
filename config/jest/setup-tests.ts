import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
