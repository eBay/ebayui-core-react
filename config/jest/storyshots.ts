import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
