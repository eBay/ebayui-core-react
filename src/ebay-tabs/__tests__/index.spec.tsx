import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayTabs, EbayTab, EbayTabPanel } from '../index'


describe('<EbayTabs>', () => {
    describe('on `tab` key press', () => {
        let wrapper, tabs

        beforeEach(() => {
            wrapper = render(
                <EbayTabs>
                    <EbayTab key="0">tab1</EbayTab>,
                    <EbayTab key="1">tab2</EbayTab>,
                    <EbayTabPanel key="0">Tab panel 1</EbayTabPanel>,
                    <EbayTabPanel key="1">Tab panel 2</EbayTabPanel>
                </EbayTabs>
            )
            tabs = wrapper.getAllByRole('tab')
            userEvent.tab()
        })
        it ('should focus on the first tab', () => {
            expect(tabs[0]).toHaveFocus()
        })
        it ('should focus on the next tab on Right Arrow key press', () => {
            fireEvent.keyDown(tabs[0], { key: 'ArrowRight' })
            expect(tabs[1]).toHaveFocus()
        })
        it ('should loop focus on the first tab on Right Arrow key press', () => {
            fireEvent.keyDown(tabs[0], { key: 'ArrowRight' })
            fireEvent.keyDown(tabs[1], { key: 'ArrowRight' })
            expect(tabs[0]).toHaveFocus()
        })
        it ('should focus on the previous tab on Left Arrow key press', () => {
            fireEvent.keyDown(tabs[0], { key: 'ArrowLeft' })
            expect(tabs[1]).toHaveFocus()
        })
    })

    describe('after losing focus', () => {
        let wrapper, tabs

        beforeEach(() => {
            wrapper = render(
                <EbayTabs>
                    <EbayTab key="0">tab1</EbayTab>,
                    <EbayTab key="1">tab2</EbayTab>,
                    <EbayTabPanel key="0">Tab panel 1</EbayTabPanel>,
                    <EbayTabPanel key="1">Tab panel 2</EbayTabPanel>
                </EbayTabs>
            )
            tabs = wrapper.getAllByRole('tab')
            userEvent.tab()
        })

        it('should refocus on the latest focused tab', () => {
            expect(tabs[0]).toHaveFocus()
            fireEvent.keyDown(tabs[0], { key: 'ArrowRight' })

            expect(tabs[1]).toHaveFocus()
            userEvent.tab()

            expect(tabs[0]).not.toHaveFocus()
            expect(tabs[1]).not.toHaveFocus()

            userEvent.tab({ shift: true })
            expect(tabs[1]).toHaveFocus()
        })
    })

    describe('on tab click', () => {
        let wrapper, spy, tabs

        beforeEach(() => {
            spy = jest.fn()
            wrapper = render(
                <EbayTabs onTabSelect={spy}>
                    <EbayTab key="0">tab1</EbayTab>,
                    <EbayTab key="1">tab2</EbayTab>,
                    <EbayTabPanel key="0">Tab panel 1</EbayTabPanel>,
                    <EbayTabPanel key="1">Tab panel 2</EbayTabPanel>
                </EbayTabs>
            )
            tabs = wrapper.getAllByRole('tab')
        })

        it('should fire an event', () => {
            fireEvent.click(tabs[0])
            expect(spy).toBeCalled()
        })

        it('should select the new tab on click', () => {
            fireEvent.click(tabs[1])

            expect(tabs[1]).not.toHaveFocus()
            expect(wrapper.getByText('Tab panel 1')).not.toBeVisible()
            expect(wrapper.getByText('Tab panel 2')).toBeVisible()
        })

        it('should change the selected/focused tab with keyboard', () => {
            expect(tabs[0]).not.toHaveFocus()

            fireEvent.click(tabs[1])
            fireEvent.keyDown(tabs[1], { key: 'ArrowLeft' })

            expect(tabs[0]).toHaveFocus()
        })
    })

    describe('in `manual` mode', () => {
        describe('on tab click', () => {
            let wrapper, spy, tabs

            beforeEach(() => {
                spy = jest.fn()
                wrapper = render(
                    <EbayTabs activation="manual" onTabSelect={spy}>
                        <EbayTab key="0">tab1</EbayTab>,
                        <EbayTab key="1">tab2</EbayTab>,
                        <EbayTabPanel key="0">Tab panel 1</EbayTabPanel>,
                        <EbayTabPanel key="1">Tab panel 2</EbayTabPanel>
                    </EbayTabs>
                )
                tabs = wrapper.getAllByRole('tab')
            })

            it('should fire an event', () => {
                fireEvent.click(tabs[0])
                expect(spy).toBeCalled()
            })

            it('should select the new tab on click', () => {
                fireEvent.click(tabs[1])

                expect(tabs[0]).not.toHaveFocus()
                expect(tabs[1]).not.toHaveFocus()
                expect(wrapper.getByText('Tab panel 1')).not.toBeVisible()
                expect(wrapper.getByText('Tab panel 2')).toBeVisible()
            })

            it('should change the selected/focused tab with keyboard', () => {
                expect(tabs[0]).not.toHaveFocus()
                expect(wrapper.getByText('Tab panel 1')).toBeVisible()

                fireEvent.click(tabs[1])
                expect(wrapper.getByText('Tab panel 2')).toBeVisible()
                expect(tabs[0]).not.toHaveFocus()
                expect(tabs[1]).not.toHaveFocus()

                fireEvent.keyDown(tabs[1], { key: 'ArrowLeft' })
                expect(wrapper.getByText('Tab panel 2')).toBeVisible()
                expect(wrapper.getByText('Tab panel 1')).not.toBeVisible()

                expect(tabs[0]).toHaveFocus()
                fireEvent.keyDown(tabs[0], { key: 'Enter' })
                expect(wrapper.getByText('Tab panel 1')).toBeVisible()

                expect(tabs[0]).toHaveFocus()
                fireEvent.keyDown(tabs[0], { key: 'ArrowLeft' })
                expect(tabs[1]).toHaveFocus()
                fireEvent.keyDown(tabs[1], { key: ' ' })
                expect(wrapper.getByText('Tab panel 2')).toBeVisible()
                expect(wrapper.getByText('Tab panel 1')).not.toBeVisible()
            })
        })

        describe('on `tab` key press', () => {
            let wrapper, tabs

            beforeEach(() => {
                wrapper = render(
                    <EbayTabs activation="manual">
                        <EbayTab key="0">tab1</EbayTab>,
                        <EbayTab key="1">tab2</EbayTab>,
                        <EbayTabPanel key="0">Tab panel 1</EbayTabPanel>,
                        <EbayTabPanel key="1">Tab panel 2</EbayTabPanel>
                    </EbayTabs>
                )
                tabs = wrapper.getAllByRole('tab')
                userEvent.tab()
            })
            it('should focus on the first tab', () => {
                expect(tabs[0]).toHaveFocus()
            })
            it('should focus on the next tab on Right Arrow key press', () => {
                fireEvent.keyDown(tabs[0], { key: 'ArrowRight' })
                expect(tabs[1]).toHaveFocus()
            })
            it('should loop focus on the first tab on Right Arrow key press', () => {
                fireEvent.keyDown(tabs[0], { key: 'ArrowRight' })
                fireEvent.keyDown(tabs[1], { key: 'ArrowRight' })
                expect(tabs[0]).toHaveFocus()
            })
            it('should focus on the previous tab on Left Arrow key press', () => {
                fireEvent.keyDown(tabs[0], { key: 'ArrowLeft' })
                expect(tabs[1]).toHaveFocus()
            })
        })

        describe('after losing focus', () => {
            let wrapper, tabs

            beforeEach(() => {
                wrapper = render(
                    <EbayTabs activation="manual">
                        <EbayTab key="0">tab1</EbayTab>,
                        <EbayTab key="1">tab2</EbayTab>,
                        <EbayTabPanel key="0">Tab panel 1</EbayTabPanel>,
                        <EbayTabPanel key="1">Tab panel 2</EbayTabPanel>
                    </EbayTabs>
                )
                tabs = wrapper.getAllByRole('tab')
                userEvent.tab()
            })

            it('should refocus on the latest focused tab', () => {
                expect(tabs[0]).toHaveFocus()
                fireEvent.keyDown(tabs[0], { key: 'ArrowRight' })

                expect(tabs[1]).toHaveFocus()
                userEvent.tab()

                expect(tabs[0]).not.toHaveFocus()
                expect(tabs[1]).not.toHaveFocus()

                userEvent.tab({ shift: true })
                expect(tabs[1]).toHaveFocus()
            })
        })
    })
})

initStoryshots({
    config: ({ configure }) =>
        configure(() => {
            require('./index.stories')
        }, module)
})
