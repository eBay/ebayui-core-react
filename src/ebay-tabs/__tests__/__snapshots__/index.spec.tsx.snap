// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-tabs Bottom Tabs on Long Page 1`] = `
<DocumentFragment>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <p
    style="margin: 10em 0px;"
  >
    Scroll down for more content...
  </p>
  <div
    class="tabs"
  >
    <div
      class="tabs__items"
      role="tablist"
    >
      <div
        aria-controls="default-tabpanel-0"
        aria-selected="true"
        class="tabs__item"
        id="default-tab-0"
        role="tab"
        tabindex="0"
      >
        <span>
          Tab 1
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-1"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-1"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 2
        </span>
      </div>
    </div>
    <div
      class="tabs__content"
    >
      <div
        aria-labelledby="default-tab-0"
        class="tabs__panel"
        id="default-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 1
          </h3>
          <p>
            Panel 1 content.
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-1"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-1"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 2
          </h3>
          <p>
            Panel 2 content.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Default Tabs 1`] = `
<DocumentFragment>
  <div
    class="tabs"
  >
    <div
      class="tabs__items"
      role="tablist"
    >
      <div
        aria-controls="default-tabpanel-0"
        aria-selected="true"
        class="tabs__item"
        id="default-tab-0"
        role="tab"
        tabindex="0"
      >
        <span>
          Tab 1
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-1"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-1"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 2
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-2"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-2"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 3
        </span>
      </div>
    </div>
    <div
      class="tabs__content"
    >
      <div
        aria-labelledby="default-tab-0"
        class="tabs__panel"
        id="default-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 1
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 1\` (skipping \`Tab 2\` and \`Tab 3\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-1"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-1"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 2
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 2\` (skipping \`Tab 3\` and \`Tab 1\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-2"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-2"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 3
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 3\` (skipping \`Tab 1\` and \`Tab 2\`)
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Fake Tabs (links) 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item fake-tabs__item--current"
      >
        <a
          aria-current="page"
          href="https://www.ebay.com/1"
        >
          Tab 1
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          href="https://www.ebay.com/2"
        >
          Tab 2
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          href="https://www.ebay.com/3"
        >
          Tab 3
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__cell"
      >
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla ornare, quam at lacinia pretium, lacus urna luctus nisi, eget molestie massa tortor id lacus. Aenean ac fringilla lacus. Fusce vel dui ex. Vivamus luctus egestas nulla, non hendrerit purus luctus at. Maecenas vel diam enim. Pellentesque quam neque, porttitor tincidunt vestibulum at, dapibus sit amet tortor.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Large Tabs 1`] = `
<DocumentFragment>
  <div
    class="tabs"
  >
    <div
      class="tabs__items tabs__items--large"
      role="tablist"
    >
      <div
        aria-controls="default-tabpanel-0"
        aria-selected="true"
        class="tabs__item"
        id="default-tab-0"
        role="tab"
        tabindex="0"
      >
        <span>
          Large Tab 1
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-1"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-1"
        role="tab"
        tabindex="-1"
      >
        <span>
          Large Tab 2
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-2"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-2"
        role="tab"
        tabindex="-1"
      >
        <span>
          Large Tab 3
        </span>
      </div>
    </div>
    <div
      class="tabs__content"
    >
      <div
        aria-labelledby="default-tab-0"
        class="tabs__panel"
        id="default-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 1
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 1\` (skipping \`Tab 2\` and \`Tab 3\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-1"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-1"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 2
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 2\` (skipping \`Tab 3\` and \`Tab 1\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-2"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-2"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 3
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 3\` (skipping \`Tab 1\` and \`Tab 2\`)
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Manually activated Tabs 1`] = `
<DocumentFragment>
  <div
    class="tabs"
  >
    <div
      class="tabs__items"
      role="tablist"
    >
      <div
        aria-controls="default-tabpanel-0"
        aria-selected="true"
        class="tabs__item"
        id="default-tab-0"
        role="tab"
        tabindex="0"
      >
        <span>
          Tab 1
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-1"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-1"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 2
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-2"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-2"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 3
        </span>
      </div>
    </div>
    <div
      class="tabs__content"
    >
      <div
        aria-labelledby="default-tab-0"
        class="tabs__panel"
        id="default-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 1
          </h3>
          <p>
            These tabs are activated manually on Space/Enter key press
          </p>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 1\` (skipping \`Tab 2\` and \`Tab 3\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-1"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-1"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 2
          </h3>
          <p>
            These tabs are activated manually on Space/Enter key press
          </p>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 2\` (skipping \`Tab 3\` and \`Tab 1\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-2"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-2"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <p>
            These tabs are activated manually on Space/Enter key press
          </p>
          <h3>
            Panel 3
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 3\` (skipping \`Tab 1\` and \`Tab 2\`)
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Multiple Tabs 1`] = `
<DocumentFragment>
  <div>
    <div
      class="tabs"
      id="tab-upper"
    >
      <div
        class="tabs__items"
        role="tablist"
      >
        <div
          aria-controls="tab-upper-tabpanel-0"
          aria-selected="true"
          class="tabs__item"
          id="tab-upper-tab-0"
          role="tab"
          tabindex="0"
        >
          <span>
            Tab 1
          </span>
        </div>
        <div
          aria-controls="tab-upper-tabpanel-1"
          aria-selected="false"
          class="tabs__item"
          id="tab-upper-tab-1"
          role="tab"
          tabindex="-1"
        >
          <span>
            Tab 2
          </span>
        </div>
      </div>
      <div
        class="tabs__content"
      >
        <div
          aria-labelledby="default-tab-0"
          class="tabs__panel"
          id="tab-upper-tabpanel-0"
          role="tabpanel"
        >
          <div
            class="tabs__cell"
          >
            <h3>
              Panel 1
            </h3>
            <p>
              Panel 1 content. Here is a 
              <a
                href="#"
              >
                link
              </a>
              .
            </p>
          </div>
        </div>
        <div
          aria-labelledby="default-tab-1"
          class="tabs__panel"
          hidden=""
          id="tab-upper-tabpanel-1"
          role="tabpanel"
        >
          <div
            class="tabs__cell"
          >
            <h3>
              Panel 2
            </h3>
            <p>
              Panel 2 content. Here is a 
              <a
                href="#"
              >
                link
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style="margin-top: 64px;"
  >
    <div
      class="tabs"
      id="tab-lower"
    >
      <div
        class="tabs__items"
        role="tablist"
      >
        <div
          aria-controls="tab-lower-tabpanel-0"
          aria-selected="true"
          class="tabs__item"
          id="tab-lower-tab-0"
          role="tab"
          tabindex="0"
        >
          <span>
            Tab A
          </span>
        </div>
        <div
          aria-controls="tab-lower-tabpanel-1"
          aria-selected="false"
          class="tabs__item"
          id="tab-lower-tab-1"
          role="tab"
          tabindex="-1"
        >
          <span>
            Tab B
          </span>
        </div>
        <div
          aria-controls="tab-lower-tabpanel-2"
          aria-selected="false"
          class="tabs__item"
          id="tab-lower-tab-2"
          role="tab"
          tabindex="-1"
        >
          <span>
            Tab C
          </span>
        </div>
      </div>
      <div
        class="tabs__content"
      >
        <div
          aria-labelledby="default-tab-0"
          class="tabs__panel"
          id="tab-lower-tabpanel-0"
          role="tabpanel"
        >
          <div
            class="tabs__cell"
          >
            <h3>
              Panel A
            </h3>
            <p>
              Panel A content. Here is a 
              <a
                href="#"
              >
                link
              </a>
              .
            </p>
          </div>
        </div>
        <div
          aria-labelledby="default-tab-1"
          class="tabs__panel"
          hidden=""
          id="tab-lower-tabpanel-1"
          role="tabpanel"
        >
          <div
            class="tabs__cell"
          >
            <h3>
              Panel B
            </h3>
            <p>
              Panel B content. Here is a 
              <a
                href="#"
              >
                link
              </a>
              .
            </p>
          </div>
        </div>
        <div
          aria-labelledby="default-tab-2"
          class="tabs__panel"
          hidden=""
          id="tab-lower-tabpanel-2"
          role="tabpanel"
        >
          <div
            class="tabs__cell"
          >
            <h3>
              Panel C
            </h3>
            <p>
              Panel C content. Here is a 
              <a
                href="#"
              >
                link
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Pre-selected Tab 1`] = `
<DocumentFragment>
  <div
    class="tabs"
  >
    <div
      class="tabs__items"
      role="tablist"
    >
      <div
        aria-controls="default-tabpanel-0"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-0"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 1
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-1"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-1"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 2
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-2"
        aria-selected="true"
        class="tabs__item"
        id="default-tab-2"
        role="tab"
        tabindex="0"
      >
        <span>
          Tab 3
        </span>
      </div>
    </div>
    <div
      class="tabs__content"
    >
      <div
        aria-labelledby="default-tab-0"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 1
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 1\` (skipping \`Tab 2\` and \`Tab 3\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-1"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-1"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 2
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 2\` (skipping \`Tab 3\` and \`Tab 1\`)
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-2"
        class="tabs__panel"
        id="default-tabpanel-2"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 3
          </h3>
          <p>
            Focus should go to 
            <a
              href="#"
            >
              this link
            </a>
             using \`tab\`-key immediately after the \`Tab 3\` (skipping \`Tab 1\` and \`Tab 2\`)
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Programmatically selected Tabs 1`] = `
<DocumentFragment>
  <button>
    Select Tab 1
  </button>
  <button>
    Select Tab 2
  </button>
  <button>
    Select Tab 3
  </button>
  <div
    class="tabs"
  >
    <div
      class="tabs__items"
      role="tablist"
    >
      <div
        aria-controls="default-tabpanel-0"
        aria-selected="true"
        class="tabs__item"
        id="default-tab-0"
        role="tab"
        tabindex="0"
      >
        <span>
          Tab 1
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-1"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-1"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 2
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-2"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-2"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 3
        </span>
      </div>
    </div>
    <div
      class="tabs__content"
    >
      <div
        aria-labelledby="default-tab-0"
        class="tabs__panel"
        id="default-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 1
          </h3>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-1"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-1"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 2
          </h3>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-2"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-2"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 3
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-tabs Tabs with inputs 1`] = `
<DocumentFragment>
  <div
    class="tabs"
  >
    <div
      class="tabs__items"
      role="tablist"
    >
      <div
        aria-controls="default-tabpanel-0"
        aria-selected="true"
        class="tabs__item"
        id="default-tab-0"
        role="tab"
        tabindex="0"
      >
        <span>
          Tab 1
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-1"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-1"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 2
        </span>
      </div>
      <div
        aria-controls="default-tabpanel-2"
        aria-selected="false"
        class="tabs__item"
        id="default-tab-2"
        role="tab"
        tabindex="-1"
      >
        <span>
          Tab 3
        </span>
      </div>
    </div>
    <div
      class="tabs__content"
    >
      <div
        aria-labelledby="default-tab-0"
        class="tabs__panel"
        id="default-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <label
            for="name"
          >
            Name 
          </label>
          <input
            id="name"
            placeholder="john doe"
            type="text"
            value="John Doe"
          />
          <p>
            My name is John Doe
          </p>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-1"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-1"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 2
          </h3>
        </div>
      </div>
      <div
        aria-labelledby="default-tab-2"
        class="tabs__panel"
        hidden=""
        id="default-tabpanel-2"
        role="tabpanel"
      >
        <div
          class="tabs__cell"
        >
          <h3>
            Panel 3
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
